lib/Parse/Lexer.cpp

```diff
@@ -1373,6 +1373,64 @@ void Lexer::lexEscapedIdentifier() {
   formToken(tok::backtick, Quote);
 }
 
+/// Find the end of a version control conflict marker.
+static const char *findConflictEnd(const char *CurPtr, const char *BufferEnd,
+                                   ConflictMarkerKind CMK) {
+  const char *terminator = CMK == ConflictMarkerKind::Perforce ? "<<<<\n"
+                                                               : ">>>>>>>";
+  size_t termLen = CMK == ConflictMarkerKind::Perforce ? 5 : 7;
+  
+  // Get a reference to the rest of the buffer minus the length of the start
+  // of the conflict marker.
+  auto restOfBuffer = StringRef(CurPtr, BufferEnd - CurPtr).substr(termLen);
+  size_t endPos = restOfBuffer.find(terminator);
+  while (endPos != StringRef::npos) {
+    // Must occur at start of line.
+    if (endPos == 0 ||
+        (restOfBuffer[endPos - 1] != '\r' && restOfBuffer[endPos - 1] != '\n'))
+    {
+      restOfBuffer = restOfBuffer.substr(endPos+termLen);
+      endPos = restOfBuffer.find(terminator);
+      continue;
+    }
+    return restOfBuffer.data() + endPos;
```

Comment:
Can you invert the logic here, so that the continue is the fall-through case? I think using the overload of `find` that takes a start location will make it easier—then you can access the previous character without having to check for 0.